components/google-cloud/tests/experimental/remote/gcp_launcher/test_model_upload_remote_runner.py

```diff
@@ -0,0 +1,138 @@
+        self._project = 'test_project'
+        self._location = 'test_region'
+        self._type = 'ModelUpload'
+        self._lro_name = f'/projects/{self._project}/locations/{self._location}/operations/123'
+        self._model_name = f'/projects/{self._project}/locations/{self._location}/models/123'
+        self._executor_input = '{"outputs":{"artifacts":{"model":{"artifacts":[{"metadata":{},"name":"foobar","type":{"schemaTitle":"system.Model"},"uri":"gs://abc"}]}},"outputFile":"localpath/foo"}}'
+        self._output_file_path = 'localpath/foo'
+        self._gcp_resouces_path = 'gcp_resouces'
+        self._uri_prefix = f"https://{self._location}-aiplatform.googleapis.com/v1/"
+
+    def tearDown(self):
+        if os.path.exists(self._gcp_resouces_path):
+            os.remove(self._gcp_resouces_path)
+
+    @mock.patch.object(aiplatform.gapic, 'ModelServiceClient', autospec=True)
+    def test_model_upload_remote_runner_succeeded(self,
+                                                  mock_model_service_client):
+        model_client = mock.Mock()
+        mock_model_service_client.return_value = model_client
+
+        upload_model_lro = mock.Mock()
+        model_client.upload_model.return_value = upload_model_lro
+        upload_model_lro.operation.name = self._lro_name
+
+        upload_model_lro.done.return_value = True
+        upload_model_lro.operation.error.code = 0
+        result = LroResult()
+        result.model = self._model_name
+        upload_model_lro.result.return_value = result
+
+        upload_model_remote_runner.upload_model(self._type, self._project,
+                                                self._location, self._payload,
+                                                self._gcp_resouces_path,
+                                                self._executor_input)
+        mock_model_service_client.assert_called_once_with(
+            client_options={
+                'api_endpoint': 'test_region-aiplatform.googleapis.com'
+            },
+            client_info=mock.ANY)
+
+        with open(self._output_file_path) as f:
+            executor_output = f.read()
+            self.assertEqual(
+                executor_output,
+                '{"artifacts": {"model": {"artifacts": [{"metadata": {}, "name": "foobar", "type": {"schemaTitle": "system.Model"}, "uri": "https://test_region-aiplatform.googleapis.com/v1//projects/test_project/locations/test_region/models/123"}]}}}'
+            )
+
+        with open(self._gcp_resouces_path) as f:
+            serialized_gcp_resources = f.read()
+            # Instantiate GCPResources Proto
+            lro_resources = json_format.Parse(serialized_gcp_resources,
+                                              GcpResources())
+
+            self.assertEqual(len(lro_resources.resources), 1)
```

Comment:
might be better to use `self.assertLen` here.

Reply:
it doesn't seem to have assertLen.
please let me know if it exists.

Reply:
ahh interesting i guess it is only in tensorflow unittest package https://www.tensorflow.org/api_docs/python/tf/test/TestCase#assertLen my bad.